src/main/scala/PSO/PSOExecutor.scala

```diff
@@ -18,26 +18,33 @@
 package com.trifectalabs.myriad
 package pso
 
+import akka.pattern.ask
+import akka.util.Timeout
+import scala.concurrent.duration._
+
 import scala.concurrent.Future
-import scala.concurrent.ExecutionContext.Implicits.global
 
 class PSOExecutor(pso: PSOSystem) extends Executor {
-  override def run(): Future[Result] = {
-    val particleCount = pso.particles.length
-
-    println(s"\nExecuting job with:")
-    println(s"\tParticle count: $particleCount")
-    println(s"\tOptimization Function: ${pso.config.objectiveFunction}")
-    println(s"\tTermination Criteria: ${pso.config.terminationCriteria}")
-
-    pso.master ! PrintSwarmScore
-
-    // termination criteria is # of iterations
-    for(i <- 1 to 1000) {
-      pso.particles.foreach(p => p ! ComputeIteration)
+  override def run: Future[Result] = {
+    // Kick off computation for each particle
+    pso.particles.foreach(_ ! ComputeIteration)
+    // TODO: make timeout variable depending on length of computation
+    implicit val timeout = Timeout(5.seconds)
+    implicit val ec = pso.system.dispatcher
+    // retrieve the near-optimal value from the swarm master
+    val best: Future[Any] = pso.master ? ResultRequest
+    // wrap value in Result
+    for {
```

Comment:
If this is the only thing in your for comprehension, you probably don't need to use a for comprehension. Map on the future instead. 

`best map { b => b match ....`